### PR TITLE
Convert more classes to CompileStatic

### PR DESCRIPTION
**Pull request by ejona86**
As a solution to use AnimalSniffer to detect incompatibilities with
Java 8, as discussed at
https://github.com/google/protobuf-gradle-plugin/pull/565#issuecomment-1159752656

CC @rougsig

### CODE REVIEW

File: src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy [hunk 1]
```diff
@@ -415,6 +418,7 @@ public abstract class GenerateProtoTask extends DefaultTask {
     return flavors
   }
 
+  @TypeChecked(TypeCheckingMode.SKIP) // Don't depend on AGP
```
Comment (rougsig) [inline src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy:421]: I'd prefer to add facade on SourceSet in some reasons:
1 better code, no Object
2 agp contains two different classes for sourceSets, deprecated one and the new one. Its the main reason why plugin doesnt work with newer agp.
3 a bit easy support to kmp

**Comment by rougsig** [comment]
@ejona86 I will add animal sniffer plugin tomorrow. Or will you do it?

File: src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy (hunk 1 above)
Comment (ejona86) [reply src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy:421]: I have little issue with that, although it seems best to do that separately. Right now it is "get as much compiling static as easily possible." This is a trivial change and gets us closer to Java 11, if not AGP. Over time we  increase what is compiled static and clean AGP handling (and fix AGP).

**Comment by ejona86** [comment]
It doesn't really matter to me who does it. I had hoped to do a pass of ProtobufPlugin to CompileStatic since it's a rather important class. But looking, it seems maybe ¾ of the class can't be converted because it mixes in Android usages. I guess we'll just leave that for later.

**Review by YifeiZhuang — APPROVED** [review]

**Pull request merged**